src/compile/data/parse.ts

```diff
@@ -86,6 +86,12 @@ export function parseTransformArray(head: DataFlowNode, model: Model, ancestorPa
       for (const field of keys(window.producedFields())) {
         ancestorParse.set(field, 'derived', false);
       }
+    } else if (isStack(t)) {
+      const stack = head = StackNode.makeFromTransform(head, t);
+
+      for (const field of keys(stack.producedFields())) {
```

Comment:
This code is called repeatedly through this method.  In a separate PR, we should refactor this part https://github.com/vega/vega-lite/issues/3772.